sdk/lib/crt/startup/atonexit.c

```diff
@@ -24,13 +24,31 @@
 _PVFV *__onexitbegin;
 _PVFV *__onexitend;
 
-extern _CRTIMP _onexit_t __cdecl __dllonexit (_onexit_t, _PVFV**, _PVFV**);
+extern _onexit_t __cdecl __dllonexit (_onexit_t, _PVFV**, _PVFV**);
 extern _onexit_t (__cdecl * __MINGW_IMP_SYMBOL(_onexit)) (_onexit_t func);
 
+/* INTERNAL: call atexit functions */
+void __call_atexit(void)
+{
+    /* Note: should only be called with the exit lock held */
+    _PVFV *first, *last;
+
+    first =  (_PVFV *)_decode_pointer(__onexitbegin);
+    last = (_PVFV *)_decode_pointer(__onexitend);;
+
+    if (!first) return;
+
+    while (--last >= first)
+        if (*last)
+            (**last)();
```

Comment:
```suggestion
    while (--last >= first)
    {
        if (*last)
            (**last)();
    }
```
unless this was coming from 3rd party code (wine...)?